thrust/mr/new.h

```diff
@@ -65,6 +69,11 @@ class new_delete_resource THRUST_FINAL : public memory_resource<>
         ::operator delete(p, bytes, std::align_val_t(alignment));
 #else
         (void)alignment;
+        
+        // If `p` is null, do nothing
+        if(nullptr == p){
```

Comment:
I wonder if this should also check for `bytes == 0`?

Reply:
I'm assuming that the value of `bytes` must be the same in `do_deallocate` as it was when `do_allocate` was called, in which case there's no need to do the additional check. Adding `assert((nullptr == p) == (0 == bytes));` might be a nice diagnostic for debug builds, though.

@griwes Do you agree?

Reply:
It's a precondition that you invoke `deallocate` with the same arguments that were passed to the `allocate` call that produced the pointer. It's fine for us to just do nothing on nullptr, assuming returning it is correct in the first place (see the comment above); it's UB if the user does something wrong here, so just always doing nothing on nullptr would be perfectly fine.